api/host_test.go

```diff
@@ -49,7 +51,115 @@ var (
 	}
 )
 
-// TestWorkingStatus tests that the host's WorkingStatus field is set correctly.
+// TestEstimateWeight tests that /host/estimatescore works correctly.
+func TestEstimateWeight(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	t.Parallel()
+
+	st, err := createServerTester(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer st.server.panicClose()
+
+	// announce a host, create an allowance, upload some data.
+	if err := st.announceHost(); err != nil {
+		t.Fatal(err)
+	}
+	if err := st.acceptContracts(); err != nil {
+		t.Fatal(err)
+	}
+	if err := st.setHostStorage(); err != nil {
+		t.Fatal(err)
+	}
+
+	var eg HostEstimateScoreGET
+	if err := st.getAPI("/host/estimatescore", &eg); err != nil {
+		t.Fatal(err)
+	}
+	originalEstimate := eg.EstimatedScore
+
+	// verify that the estimate is being correctly updated by setting a massively
+	// increased min contract price and verifying that the score decreases.
+	is := st.host.InternalSettings()
+	is.MinContractPrice = is.MinContractPrice.Add(types.SiacoinPrecision.Mul64(9999999999))
+	if err := st.host.SetInternalSettings(is); err != nil {
+		t.Fatal(err)
+	}
+	if err := st.getAPI("/host/estimatescore", &eg); err != nil {
+		t.Fatal(err)
+	}
+	if eg.EstimatedScore.Cmp(originalEstimate) != -1 {
+		t.Fatal("score estimate did not decrease after incrementing mincontractprice")
+	}
+	if eg.ConversionRate != float64(100) {
+		t.Fatal("incorrect conversion rate", eg.ConversionRate)
+	}
+
+	// add a few hosts to the hostdb and verify that the conversion rate is
+	// reflected correctly
+	st2, err := blankServerTester(t.Name() + "-st2")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer st2.panicClose()
+	st3, err := blankServerTester(t.Name() + "-st3")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer st3.panicClose()
+	st4, err := blankServerTester(t.Name() + "-st4")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer st4.panicClose()
+	sts := []*serverTester{st, st2, st3, st4}
+	err = fullyConnectNodes(sts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = fundAllNodes(sts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for i, tester := range sts {
+		is = tester.host.InternalSettings()
+		is.MinContractPrice = types.SiacoinPrecision.Mul64(1000 + (1000 * uint64(i)))
+		err = tester.host.SetInternalSettings(is)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	err = announceAllHosts(sts)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		price          types.Currency
+		conversionRate float64
+	}{
+		{types.SiacoinPrecision, 100},
+		{types.SiacoinPrecision.Mul64(30000), 98},
+		{types.SiacoinPrecision.Mul64(50000), 96},
+		{types.SiacoinPrecision.Mul64(70000), 94},
+		{types.SiacoinPrecision.Mul64(60000000), 94},
```

Comment:
I'm assuming this test will break if we change the scoring algo. Kind of annoying, but not much we can do about it for now.